src/tags/models.py

```diff
@@ -1,8 +1,27 @@
 from django.db import models
 
+from django.apps import apps
 from taggit.managers import TaggableManager
 
 
+def total_xp_by_tags(user, tags: list):
+    """ 
+        Returns user's total xp over active_semester filtered by tags
+    """ 
+    # get models through here to prevent circular imports
+    QuestSubmission = apps.get_model("quest_manager", "QuestSubmission")
+    BadgeAssertion = apps.get_model("badges", "BadgeAssertion")
+
+    # get all objects with xp that is related to user
+    # convert submission/assertion qs into Quest/Badge qs
+    xp_list = [
+        [qs.quest.xp * qs.ordinal for qs in QuestSubmission.objects.all_approved(user=user).filter(quest__tags__name__in=tags).distinct()], 
```

Comment:
I  don't think ordinal will work here.  For 2 reasons:
1. ordinals don't reset each semester, so for example, if a student submits only one Tinkering (10 XP) quest in a new semester, but has submitted nine last course/semester, will this count it as 100 xp? (I think semester - only xp is not considered here actually.  I can't remember if all_approved defaults to current semester only or not, same with Badge all for user).  There may already be a method on the Quest or QuestSubmission model that counts how many of that quest the student has completed (for this same problem), if you want to continue with this approach.
2. ordinals don't indicate how many quests they've completed, only how many they's started.  Say they have Tinkering (1) and Tinkering (2) and Tinkering (3), it's possible Tinkering (2) could be returned, or even dropped.  So now they've actually only got XP for Tinkering (1) and (3). As mentioned above, I think there is a method that gets you this already...I hope there is anyway =)

Reply:
So ordinals are just cosmetic, as they are meant to tell the user how many of a quest/badge they have?

If thatst the case then fixed

Reply:
Ya, basically cosmetic, it's to be able to differentiate the different submissions.  It's also the number of that quest they started (but not necessarily the number they've submitted or had approved, etc)